cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -863,4 +871,42 @@ private void handleAuthException(Runnable function) {
     public Completable deleteGenericResourceByIdAsync(String databaseServerId) {
         return handleAuthException(() -> azure.genericResources().deleteByIdAsync(databaseServerId));
     }
+
+    private DiskEncryptionSetInner createDiskEncryptionSetInner(String sourceVaultId, String encryptionKeyUrl, String location, Map<String, String> tags) {
+        SourceVault sourceVault = new SourceVault().withId(sourceVaultId);
+        KeyVaultAndKeyReference keyUrl = new KeyVaultAndKeyReference().withKeyUrl(encryptionKeyUrl).withSourceVault(sourceVault);
+        EncryptionSetIdentity eSetId = new EncryptionSetIdentity().withType(DiskEncryptionSetIdentityType.SYSTEM_ASSIGNED);
+        return (DiskEncryptionSetInner) new DiskEncryptionSetInner()
```

Comment:
why we need this cast?

Reply:
`.withLocation()` and `.withTags()` are actually inherited from the superclass of `DiskEncryptionSetInner`, `com.microsoft.azure.Resource`, and they return that type instead of `DiskEncryptionSetInner`. We could get away with this cast by explicitly storing the `DiskEncryptionSetInner` in a local variable (up to the `.withIdentity()` invocation) and calling `.withLocation()` and `.withTags()` on that variable separately.